src/Workspaces/CSharp/Portable/Workspace/LanguageServices/CSharpSyntaxTreeFactoryService.RecoverableSyntaxTree.cs

```diff
@@ -143,19 +134,20 @@ public override SyntaxReference GetReference(SyntaxNode node)
                     }
                 }
 
-                CompilationUnitSyntax IRecoverableSyntaxTree<CompilationUnitSyntax>.CloneNodeAsRoot(CompilationUnitSyntax root)
+                CSharpSyntaxNode IRecoverableSyntaxTree<CSharpSyntaxNode>.CloneNodeAsRoot(CSharpSyntaxNode root)
                     => CloneNodeAsRoot(root);
 
                 public bool ContainsDirectives => _info.ContainsDirectives;
 
                 public override SyntaxTree WithRootAndOptions(SyntaxNode root, ParseOptions options)
                 {
-                    if (ReferenceEquals(_info.Options, options) && this.TryGetRoot(out var oldRoot) && ReferenceEquals(root, oldRoot))
+                    if (ReferenceEquals(_info.Options, options) && TryGetRoot(out var oldRoot) && ReferenceEquals(root, oldRoot))
                     {
                         return this;
                     }
 
-                    return Create((CSharpSyntaxNode)root, this.Options, _info.FilePath);
```

Comment:
So we just never created a recoverable tree further from here???

Reply:
Right.